compiler/rustc_trait_selection/src/traits/specialize/specialization_graph.rs

```diff
@@ -275,7 +275,7 @@ impl<'tcx> GraphExt<'tcx> for Graph {
     ) -> Result<Option<FutureCompatOverlapError<'tcx>>, OverlapError<'tcx>> {
         assert!(impl_def_id.is_local());
 
-        let trait_ref = tcx.impl_trait_ref(impl_def_id).unwrap();
+        let trait_ref = tcx.bound_impl_trait_ref(impl_def_id).unwrap().subst_identity();
```

Comment:
:shrug: conceptionally we actually keep the `EarlyBinder` here. So maybe `skip_binder` with a "FIXME: use `EarlyBinder` in `self.children`"?